R/DataManagement.R

```diff
@@ -1,9 +1,155 @@
+#' @param ... other parameters passed on to internal functions
+#' 
+#' @return The adjusted database as a data.table
+#' @export
+#' 
+checkBase <- function(base,
+                      deleteMissing = "no",
+                      convertDates = FALSE,
+                      dateFormat = "",
+                      bestGuess = FALSE,
+                      overnight = FALSE,
+                      deleteErrors = "",
+                      patientID = "pID",
+                      hospitalID = "hID",
+                      disDate = "Ddate",
+                      admDate = "Adate",
+                      ...)
+{
+    new_base = checkFormat(base,
+                           deleteMissing = deleteMissing)
+    
+    new_base = checkDates(base = new_base,
+                          convertDates = convertDates,
+                          dateFormat = dateFormat,
+                          bestGuess = bestGuess,
+                          overnight = overnight,
+                          deleteErrors = deleteErrors,
+                          ...)
+    
+    new_base = adjust_overlapping_stays(base = new_base,
+                                        patientID = "pID",#ID
+                                        hospitalID = "hID",#FINESS
+                                        admDate = "Adate",
+                                        disDate = "Ddate",
+                                        maxIteration = 25,
+                                        ...)
+    return(new_base)
+}
+
+#' Check database format
+#'
+#' Function that performs various generic checks to ensure that the database has the correct format
+#'
+#' @param base (data.table)
+#'     A patient discharge database, in the form of a data.table. The data.table should have at least the following columns:
+#'         pID: patientID (character)
+#'         hID: hospitalID (character)
+#'         Adate: admission date (POSIXct, but character can be converted to POSIXct)
+#'         Ddate: discharge date (POSIXct, but character can be converted to POSIXct)
+#' @param deleteMissing (character) How to handle records that contain a missing value in at least one of the four mandatory variables:
+#' "no" (default): do not delete. Stops the function with an error message.
+#' "record": deletes just the incorrect record.
+#' "patient": deletes all records of each patient with one or more incorrect records.
+#' 
+#' @return Returns either an error message, or the database (modified if need be).
+#' @export
+#' 
+checkFormat <- function(base,
+                        deleteMissing = "no")
+{
+    if (!"data.frame" %in% class(base)) {
+        stop("The database must be either a data.frame or a data.table object")
+    } else if (!"data.table" %in% class(base)) {
+        setDT(base)
+        message("Converting database to a data.table object")
+    }
+    
+    # Check missing columns
```

Comment:
It makes more sense to me to have that here in a separate function rather than in "checkDates()"